src/cockpit/channels/pcp.py

```
@@ -0,0 +1,672 @@
+                    raise ChannelError('') from None
+
+                assert metric_desc is not None
+                archive.metric_descriptions.append(metric_desc)
+
+        return archives
+
+    def convert_metric_description(self, context: 'pmapi.pmContext', metric: JsonObject) -> MetricInfo:
+        name = get_str(metric, 'name', '')
+        if name == '':
+            raise ChannelError('protocol-error',
+                               message='invalid "metrics" option was specified (no name for metric)')
+        units = get_str(metric, 'units', '')
+        derive = get_str(metric, 'derive', '')
+
+        try:
+            pm_ids = context.pmLookupName(name)
+        except pmapi.pmErr as exc:
+            if exc.errno() == c_api.PM_ERR_NAME:
+                logger.error("no such metric: %s", name)
+                raise MetricNotFoundError('error', message=f'no such metric: {name}') from None
+            else:
+                raise ChannelError('internal-error', message=str(exc)) from None
+
+        try:
+            pm_desc = context.pmLookupDesc(pm_ids[0])
+        except pmapi.pmErr as exc:
+            if exc.errno() == c_api.PM_ERR_NAME:
+                raise ChannelError('not-found', message=f'no such metric: {name}') from None
+            else:
+                raise ChannelError('internal-error', message=str(exc)) from None
+
+        # Multi-instance metrics have a domain defined
+        instanced = pm_desc.indom != c_api.PM_INDOM_NULL
+
+        if instanced:
+            if len(self.instances) > 0:
+                context.pmDelProfile(pm_desc, None)
+                for instance in self.instances:
+                    try:
+                        instid = context.pmLookupInDom(pm_desc, instance)
+                        context.pmAddProfile(pm_desc, instid)
+                    except pmapi.pmErr as exc:
+                        logger.debug("Unable to add profile: instance=%s err=%s", instance, exc)
```

What is a "profile"? is that expected to be opportunistic, i.e. fail silently? I.e. if you specify an instance which doesn't exist, it is simply not reported in the channel, as opposed to raising an error code?

---

OK!